fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java

```diff
@@ -912,6 +913,7 @@ protected RuntimeException mapAsyncToSyncException(@Nonnull Throwable ex) {
             } catch (TimeoutException ex) {
                 if (timer != null) {
                     timer.recordTimeout(event, startTime);
+                    throw asyncToSyncExceptionMapper.apply(new LoggableTimeoutException(ex, "TimeLimit", timeout.getLeft(), "TimeUnit", timeout.getRight()), event);
```

Comment:
Minor, but our "convention" for record layer logging keys is to be `lower_case` rather than `CamelCase` and, although we aren't really re-using these strings today, you may want to add `LogMessageKeys.TIME_LIMIT` and `LogMessageKeys.TIME_UNIT`.  At some point we really need to write down guidelines for exceptions and logging.